### ISSUE TITLE
Sox error message: Not enough space

### ISSUE THREAD

**Issue by AugustineTan** [HPCE/hpce-2014-cw2#3]
I ran the command to test that sox and lame have been installed properly:  
 `curl http://cas.ee.ic.ac.uk/people/dt10/teaching/2013/hpce/cw2/bn9th_m4.mp3 \
    | local/bin/lame -  --mp3input --decode  - \
    | local/bin/sox  - -d`

and received the following error message and output:

```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
input:  <stdin>  (44.1 kHz, 2 channels, MPEG-1 Layer III):--:-- --:--:--     0
output: <stdout>  (16 bit, Microsoft WAVE)
skipping initial 1105 samples (encoder+decoder delay)
skipping final 1066 samples (encoder padding-decoder delay)
Frame#    17/52707   96 kbps   MS    D:\Github\hpce-2014-cw2\local\bin\sox.exe FAIL sox: glob: Not enough space
Frame# 52506/52707   96 kbps   MS              0:00:04  0:00:04 --:--:-- 3848k
Frame# 52707/52707   32 kbps   MS
```

The command runs for a few seconds and then stops without playing any audio.

**Comment by m8pple** [comment]
Hrmm, it looks like it is decoding it all the way down to the last frame, then complaining
about lack of disk space. My suspicion is that it is decoding the file completely, then
when it tries to write it there is not enough space (it is about 250MB raw).

One question is, does sox appear to support the "-d" flag? For me, if I
do "local/bin/sox --help", then it lists  the "-d" flag (amongst others). If
that isn't supported (not sure why it wouldn't be), then it  would end up
writing the output to a file called "-d". Is there a large file called "-d",
or indeed any other large file that has appeared?

Alternatively, can you see any disks that have run out of space?

If that doesn't help, it looks like this is on cygwin. Some questions are:
- What type of disk is "D:" (is it a local disk, or a USB stick, or a home directory)?
- Where did the cygwin come from, is it your own install, or is it on a college machine (some have it
  others don't).
- Where are the cygwin mount points (e.g. do `mount` and look at the output)

**Comment by AugustineTan** [comment]
This problem is strange. I reached home and ran the same command again. Surprisingly the error is gone! However, my initial problem is still there in that I do not have any audio. Would you still be available at 12noon tomorrow? I believe its best we debug this in person.

Also running
`time ( ./mp3_file_src.sh bn9th_m4.mp3  > /dev/null )`
has the exact same run time as
`curl http://cas.ee.ic.ac.uk/people/dt10/teaching/2013/hpce/cw2/bn9th_m4.mp3 \ | local/bin/lame - --mp3input --decode - \ | local/bin/sox - -d`

somehow it does not seem to be decoding at the audio sampling frequency as mention in your README.

**Comment by AugustineTan** [comment]
An update on the issue. After looking at what was happening it was deduced that the error was caused by the auto "globbing" of the sox package. By disabling this feature the command would run as expected.
`
curl http://cas.ee.ic.ac.uk/people/dt10/teaching/2013/hpce/cw2/bn9th_m4.mp3 \ | local/bin/lame - --mp3input --decode - \ | local/bin/sox --no-glob - -d
`

Further investigation was carried out by running a clean build of win32 sox.exe from the website and it was found that this would run properly without the need of the additional --no-glob option. Leading me to assume that somehow my build of sox through the installation process is somehow broken.

I have tried a clean install but it still gives the same result.

**Comment by m8pple** [comment]
I've tried to replicate on my 32 and 64 bit cygwin test setups, and it seems to
play happily, so I can't reproduce the behaviour we saw.

It seemed to working fine with the `--no-glob` option, so you can just go ahead
and include that in your scripts - it won't affect the test setup, as it doesn't
use any file patterns.

So I'll update the readme to point here just in case, and if you can confirm that
it works as a solution then we can close this issue.

**Comment by AugustineTan** [comment]
Up to the section on raw data this fix seems to work fine and allows me to run everything as it should. If there are any more errors as I go along the coursework relating to this I will let you know. Thank you!

**Comment by AugustineTan** [comment]
Reopened the issue to post an update. 

I have found the cause of this problem. What helped me to find the problem was `make passthrough`. The executable file produced did not run properly with my fix from above which led me to believe there is a problem with the C compiler being used. After some poking around I found out that bash was using gcc from another MinGW installation I had on my computer. Therefore, I went to install the packages under the Devel category in Cygwin and remade all my files. This managed to fix the issue and I no  longer need `--no-glob`.
